kraken-build/src/kraken/std/cargo/__init__.py

```diff
@@ -232,6 +237,42 @@ def cargo_deny(
     )
 
 
+def cargo_hack(
+    *,
+    project: Project | None = None,
+    features: CargoHackFeatures = CargoHackFeatures.EACH,
+    action: CargoHackAction = CargoHackAction.CHECK,
+    error_message: str | None = None,
+    group: str | None = "test",
+    name: str | None = None,
+) -> CargoHackTask:
+    """Adds a task running cargo-hack for cargo projects. This ensures that all
+    combinations of the features declared in the Cargo manifests build
+    cleanly.
+
+    :param features: Method to select the features to test for, possible values
+        are CargoHackFeatures.EACH and CargoHackFeatures.POWERSET.
+    :param action: Action to run for every selected set of features, possible
+        values are CargoHackAction.BUILD, CargoHackAction.CHECK and
+        CargoHackAction.TEST.
+    """
+
+    project = project or Project.current()
+    task = project.do(
+        "cargoHack" if name is None else name,
+        CargoHackTask,
+        features=features,
+        action=action,
+        error_message=error_message,
+        group=group,
+    )
```

Comment:
```suggestion
    task = project.task(name, CargoHackTask)
    task.features = features
    task.action = action
    task.error_message = error_message
    task.group = group
```

`Project.do()` is actually deprecated in favor of `Project.task()`. This is type safe as opposed to the `do()` variant!

If you give `name` a default value you don't need to add the `is None` check.